app/src/main/java/cm/aptoide/pt/store/view/GridStoreMetaWidget.java

```diff
@@ -139,25 +139,30 @@ public GridStoreMetaWidget(View itemView) {
   }
 
   private void showBadge(HomeMeta.Badge badge, boolean storeOwner) {
-    switch (badge) {
-      case NONE:
-        badgeIcon.setImageResource(R.drawable.tin);
-        break;
-      case BRONZE:
-        badgeIcon.setImageResource(R.drawable.bronze);
-        break;
-      case SILVER:
-        badgeIcon.setImageResource(R.drawable.silver);
-        break;
-      case GOLD:
-        badgeIcon.setImageResource(R.drawable.gold);
-        break;
-      case PLATINUM:
-        badgeIcon.setImageResource(R.drawable.platinum);
-        break;
+    if (badge == null) {
+      badgeIcon.setVisibility(View.GONE);
+    } else {
+      badgeIcon.setVisibility(View.VISIBLE);
+      switch (badge) {
+        case NONE:
+          badgeIcon.setImageResource(R.drawable.tin);
+          break;
+        case BRONZE:
+          badgeIcon.setImageResource(R.drawable.bronze);
+          break;
+        case SILVER:
+          badgeIcon.setImageResource(R.drawable.silver);
+          break;
+        case GOLD:
+          badgeIcon.setImageResource(R.drawable.gold);
+          break;
+        case PLATINUM:
+          badgeIcon.setImageResource(R.drawable.platinum);
+          break;
+      }
+      badgeIcon.setOnClickListener(v -> badgeDialogFactory.create(badge, storeOwner)
+          .show());
```

Comment:
Regarding the last comment instead of null checking there would be a case NONE: which would hide the badge and a case DEFAULT: that would show tin.